tests/foreman/ui_airgun/test_contentview.py

```diff
@@ -1571,6 +1569,210 @@ def test_positive_remove_cv_version_from_default_env(session, module_org):
+        cvv_table = session.contentview.search_version(cv['name'], VERSION)
+        assert ' '.join((dev_lce.name, qe_lce.name)) == cvv_table[0]['Environments']
+
+
+@tier2
+def test_positive_remove_cv_version_from_env(session, module_org):
+    """Remove promoted content view version from environment
+
+    :id: d1da23ee-a5db-4990-9572-1a0919a9fe1c
+
+    :Steps:
+
+        1. Create a content view
+        2. Add a yum repo and a puppet module to the content view
+        3. Publish the content view
+        4. Promote the content view version to multiple environments
+            Library -> DEV -> QE
+        5. remove the content view version from QE environment
+        6. Assert: content view version exists only in Library, DEV and not in QE
+        7. Promote again from DEV -> QE
+
+    :expectedresults: Content view version exist in Library, DEV, QE
+
+    :CaseLevel: Integration
+    """
+    dev_lce = entities.LifecycleEnvironment(organization=module_org).create()
+    qe_lce = entities.LifecycleEnvironment(organization=module_org, prior=dev_lce).create()
+    puppet_module_name = 'generic_1'
+    author = 'robottelo'
+    repos = RepositoryCollection(
+        repositories=[
+            YumRepository(url=FAKE_0_YUM_REPO),
+            PuppetRepository(
+                url=CUSTOM_PUPPET_REPO,
+                modules=[dict(name=puppet_module_name, author=author)]
+            )
+        ]
+    )
+    repos.setup(module_org.id)
+    yum_repo_name = [
+        repo['name'] for repo in repos.repos_info if repo['content-type'] == 'yum'][0]
+    cv, lce = repos.setup_content_view(module_org.id, dev_lce.id)
+    cvv = entities.ContentView(id=cv['id']).read().version[0]
+    promote(cvv, qe_lce.id)
+    with session:
+        cvv = session.contentview.read_version(cv['name'], VERSION)
+        assert cvv['yum_repositories']['table'][0]['Name']
+        assert puppet_module_name in cvv['puppet_modules']['table'][0]['Name']
+        assert yum_repo_name == cvv['yum_repositories']['table'][0]['Name']
+        cvv = session.contentview.search_version(cv['name'], VERSION)[0]
+        assert qe_lce.name in cvv['Environments']
+        # remove the content view version from QE Environment
+        session.contentview.remove_version(cv['name'], VERSION, False, [qe_lce.name])
+        cvv = session.contentview.search_version(cv['name'], VERSION)[0]
+        assert qe_lce.name not in cvv['Environments']
+        # promote again to QE
+        result = session.contentview.promote(cv['name'], VERSION, qe_lce.name)
+        assert 'Promoted to {}'.format(qe_lce.name) in result['Status']
+        cvv = session.contentview.read_version(cv['name'], VERSION)
+        assert puppet_module_name in cvv['puppet_modules']['table'][0]['Name']
+        assert yum_repo_name == cvv['yum_repositories']['table'][0]['Name']
+        cvv = session.contentview.search_version(cv['name'], VERSION)[0]
+        assert qe_lce.name in cvv['Environments']
```

Comment:
Think we are not asserting the same way as required:
```
 5. remove the content view version from QE environment
 6. Assert: content view version exists only in Library, DEV and not in QE
 7. Promote again from DEV -> QE

```

Reply:
That is redundant, but I will change test case to be the same with docstring